### ISSUE TITLE
Streaming and chunked responses

### ISSUE THREAD

**Issue by zimbatm** [excon/excon#350]
Right now the response_stream only gets full chunks of size :chunk_size. This is an issue when trying to follow a continuous stream (like a stream of log lines) because in that case you want to get the latest chunk of data that arrived to forward it an not buffer it.

When going trough the code to find out the behavior I was also mislead by this argument parameter: https://github.com/geemus/excon/blob/master/lib/excon/socket.rb#L31 . `max_length` really should be called `length` if it's never returning more or less than the given size.

Also what's the point of using non-blocking IO if the control is blocked by the while loop ? In a single thread I don't see what advantage it's going to offer.

Ideally the response body would be a fake IO stream that can be read until the end of the response body (the connection can still be kept for keep-alive/pipelining) instead of using a block. That way the use can choose how to read the data and if he wants to block or not on it.

It would also solve another issue where the response_block is not called when the stream ends, giving no opportunity for the block to know when he can stop forwarding the data: https://github.com/geemus/excon/blob/master/lib/excon/response.rb#L73
